### PR TITLE
Fakeroot feature documentation

### LINKED ISSUE #126: --fakeroot flag needs to be documented
Singularity 3.3 will introduce the `--fakeroot` flag that will allow user to pretend to elevate privs inside of a container during action commands and during the build process.  This needs to be documented.  

@cclerget and @rherban

### PR DESCRIPTION
**Pull request by cclerget**
## This fixes or addresses the following GitHub issues:

- Fixes #126

### CODE REVIEW

File: fakeroot.rst [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+================
+Fakeroot feature
+================
+
+--------
+Overview
+--------
+
+Fakeroot feature (or commonly referred as rootless mode) allow an unprivileged user
```
Comment (jchavezlavalle) [inline fakeroot.rst]: allow -> allows

File: fakeroot.rst [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+to run a container as a **"fake root"** user by leveraging
+`user namespace UID/GID mapping <http://man7.org/linux/man-pages/man7/user_namespaces.7.html>`_.
+
+.. note:: 
+
+	This feature requires a Linux kernel >= 3.8, but the recommended version is >= 3.18
+
+A **"fake root"** user has almost the same administrative rights as root but only **inside the container**
+and the **requested namespaces**, which means this user:
```
Comment (jchavezlavalle) [inline fakeroot.rst]: which means **that** this user

File: fakeroot.rst [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+---------------------
+Restrictions/security
+---------------------
+
+Filesystem
+==========
+
+A **"fake root"** user can't access or modify files and directories for which he doesn't
+have already access or rights on host filesystem, so a **"fake root"** user won't be able
```
Comment (jchavezlavalle) [inline fakeroot.rst]: on **the** host filesystem

File: fakeroot.rst [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+to access to host file like ``/etc/shadow`` or host ``/root`` directory.
+
+Additionally all files or directories created with **"fake root"** user are owned like
```
Comment (jchavezlavalle) [inline fakeroot.rst]: add comma after "Additionally"

File: fakeroot.rst [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+``root:root`` inside container and are owned like ``user:group`` outside of container.
+Let's take an example, `user` is authorized to use the fakeroot feature and can use 65536
```
Comment (jchavezlavalle) [inline fakeroot.rst]: Let's see the following example, in this case "user" is authorized ....

File: fakeroot.rst [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+| 1 (daemon)           | 131072 (non-existent) |
++----------------------+-----------------------+
+| 2 (bin)              | 131073 (non-existent) |
++----------------------+-----------------------+
+| ...                  | ...                   |
++----------------------+-----------------------+
+| 65536                | 196607                |
++----------------------+-----------------------+
+
+Which means if **"fake root"** user create a file a ``bin`` user in container, this file will
```
Comment (jchavezlavalle) [inline fakeroot.rst]: create -> creates

File: fakeroot.rst [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+be owned by ``131073:131073`` outside of container. This is the responsibility of administrator to
```
Comment (jchavezlavalle) [inline fakeroot.rst]: The responsibility relies on the administrator to ensure that there is no overlap with the current user's UID/GID on the system.

File: fakeroot.rst [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+ensure there is no overlap with current user's UID/GID on the system.
+
+Network
+=======
+
+Restriction are also applied for network, if ``singularity`` is executed without ``--net`` flag,
```
Comment (jchavezlavalle) [inline fakeroot.rst]: Restriction -> Restrictions

File: fakeroot.rst [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+the **"fake root"** user won't be able to use ``ping`` or bind a container service on a port below
+than 1024.
+
+With ``--net`` the **"fake root"** user have full privileges in this dedicated network, inside
```
Comment (jchavezlavalle) [inline fakeroot.rst]: the "fake root" user **has** full privileges..

File: fakeroot.rst [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+container network he can bind  on privileged ports below 1024, use ping, manage firewall rules,
```
Comment (jchavezlavalle) [inline fakeroot.rst]: inside **the** container network he can bind on privileged ports below 1024.. (remove the space between word "bind" and "on")

File: fakeroot.rst [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+listen trafic ...
```
Comment (jchavezlavalle) [inline fakeroot.rst]: **traffic**

File: fakeroot.rst [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+  - ``shell``
+  - ``exec``
+  - ``run``
+  - ``instance start``
+  - ``build``
+
+Build
+=====
+
+With fakeroot a unprivileged user can now build image from a definition file with few restrictions, bootstrap
```
Comment (jchavezlavalle) [inline fakeroot.rst]: **an** unprivileged user

**Review by jchavezlavalle — CHANGES_REQUESTED** [review]
some small suggestions :+1: thanks @cclerget this was clearly needed

**Comment by cclerget** [comment]
Fixed. Thanks for the correction/suggestions @devprincess :+1:

**Comment by rherban** [comment]
This also fixes sylabs/singularity#3752

**Review by jchavezlavalle — APPROVED** [review]

**Pull request merged**